types/models.d.ts

```diff
@@ -303,7 +303,8 @@ declare module 'mongoose' {
       callback?: Callback<(HydratedDocument<T, TMethodsAndOverrides, TVirtuals>)[]>): Promise<Array<HydratedDocument<T, TMethodsAndOverrides, TVirtuals>>>;
     populate(doc: any, options: PopulateOptions | Array<PopulateOptions> | string,
       callback?: Callback<HydratedDocument<T, TMethodsAndOverrides, TVirtuals>>): Promise<HydratedDocument<T, TMethodsAndOverrides, TVirtuals>>;
-
+    populate<Paths={}>(docs: Array<any>, options: PopulateOptions | Array<PopulateOptions> | string): Promise<MergeType<this, Paths>>;
```

Comment:
If you had actually tested the populate doc result, you'd see that `this` is the wrong type to use here. In this context, `this` is the Model class, not the document. Use the other `populate()` calls on `Model` as an example for how to do this correctly.